test/StringToolsTests.h

```diff
@@ -0,0 +1,333 @@
+        your distribution
+
+   These and other limitations could cause serious legal problems if you ignore
+   them, so it is best to simply contact us or the Free Software Foundation, if
+   you have any questions.
+
+   Joseph Toppi - toppij@gmail.com
+   John Blackwood - makoenergy02@gmail.com
+*/
+#ifndef Mezz_Foundation_StringToolsTests_h
+#define Mezz_Foundation_StringToolsTests_h
+
+/// @file
+/// @brief This file tests the common String manipulation utilities of the Mezzanine.
+
+#include "MezzTest.h"
+
+#include "StringTools.h"
+
+DEFAULT_TEST_GROUP(StringToolsTests,StringTools)
+{
+    using namespace Mezzanine;
+
+    {//Character Checks
+        TEST_EQUAL("IsSpace(const_Char8)-Pass",true,StringTools::IsSpace(' '));
+        TEST_EQUAL("IsSpace(const_Char8)-Fail",false,StringTools::IsSpace('\t'));
+        TEST_EQUAL("IsTab(const_Char8)-Pass1",true,StringTools::IsTab('\t'));
+        TEST_EQUAL("IsTab(const_Char8)-Pass2",true,StringTools::IsTab('\v'));
+        TEST_EQUAL("IsTab(const_Char8)-Fail",false,StringTools::IsTab('\r'));
+        TEST_EQUAL("IsNewline(const_Char8)-Pass1",true,StringTools::IsNewline('\r'));
+        TEST_EQUAL("IsNewline(const_Char8)-Pass2",true,StringTools::IsNewline('\n'));
+        TEST_EQUAL("IsNewline(const_Char8)-Fail",false,StringTools::IsNewline('!'));
+        TEST_EQUAL("IsWhitespace(const_Char8)-Pass1",true,StringTools::IsWhitespace(' '));
+        TEST_EQUAL("IsWhitespace(const_Char8)-Pass2",true,StringTools::IsWhitespace('\t'));
+        TEST_EQUAL("IsWhitespace(const_Char8)-Pass3",true,StringTools::IsWhitespace('\n'));
+        TEST_EQUAL("IsWhitespace(const_Char8)-Fail",false,StringTools::IsWhitespace('&'));
+        TEST_EQUAL("IsDigit(const_Char8)-Pass",true,StringTools::IsDigit('5'));
+        TEST_EQUAL("IsDigit(const_Char8)-Fail",false,StringTools::IsDigit('L'));
+        TEST_EQUAL("IsLowerAlphaLetter(const_Char8)-Pass",true,StringTools::IsLowerAlphaLetter('a'));
+        TEST_EQUAL("IsLowerAlphaLetter(const_Char8)-Fail",false,StringTools::IsLowerAlphaLetter('A'));
+        TEST_EQUAL("IsUpperAlphaLetter(const_Char8)-Pass",true,StringTools::IsUpperAlphaLetter('B'));
+        TEST_EQUAL("IsUpperAlphaLetter(const_Char8)-Fail",false,StringTools::IsUpperAlphaLetter('b'));
+        TEST_EQUAL("IsAlphaLetter(const_Char8)-Pass",true,StringTools::IsAlphaLetter('y'));
+        TEST_EQUAL("IsAlphaLetter(const_Char8)-Fail",false,StringTools::IsAlphaLetter('9'));
+        TEST_EQUAL("IsLowerHexLetter(const_Char8)-Pass",true,StringTools::IsLowerHexLetter('e'));
+        TEST_EQUAL("IsLowerHexLetter(const_Char8)-Fail",false,StringTools::IsLowerHexLetter('g'));
+        TEST_EQUAL("IsUpperHexLetter(const_Char8)-Pass",true,StringTools::IsUpperHexLetter('F'));
+        TEST_EQUAL("IsUpperHexLetter(const_Char8)-Fail",false,StringTools::IsUpperHexLetter('1'));
+        TEST_EQUAL("IsHexLetter(const_Char8)-Pass",true,StringTools::IsHexLetter('B'));
```

Comment:
Should add a lowercase 'b' test expecting a true.

Reply:
Added.